pkg/kubectl/cmd/edit.go

```diff
@@ -145,139 +145,142 @@ func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
+			if err := results.header.writeTo(buf); err != nil {
+				return preservedFile(err, results.file, out)
+			}
+			if err := printer.PrintObj(obj, buf); err != nil {
+				return preservedFile(err, results.file, out)
+			}
+			original := buf.Bytes()
 
-		// TODO: add an annotating YAML printer that can print inline comments on each field,
-		//   including descriptions or validation errors
-
-		// generate the file to edit
-		buf := &bytes.Buffer{}
-		if err := results.header.writeTo(buf); err != nil {
-			return preservedFile(err, results.file, out)
-		}
-		if err := printer.PrintObj(obj, buf); err != nil {
-			return preservedFile(err, results.file, out)
-		}
-		original := buf.Bytes()
-
-		// launch the editor
-		edit := editor.NewDefaultEditor()
-		edited, file, err := edit.LaunchTempFile("kubectl-edit-", ext, buf)
-		if err != nil {
-			return preservedFile(err, results.file, out)
-		}
+			// launch the editor
+			edit := editor.NewDefaultEditor()
+			edited, file, err := edit.LaunchTempFile("kubectl-edit-", ext, buf)
+			if err != nil {
+				return preservedFile(err, results.file, out)
+			}
 
-		// cleanup any file from the previous pass
-		if len(results.file) > 0 {
-			os.Remove(results.file)
-		}
+			// cleanup any file from the previous pass
+			if len(results.file) > 0 {
+				os.Remove(results.file)
+			}
 
-		glog.V(4).Infof("User edited:\n%s", string(edited))
-		fmt.Printf("User edited:\n%s", string(edited))
-		lines, err := hasLines(bytes.NewBuffer(edited))
-		if err != nil {
-			return preservedFile(err, file, out)
-		}
-		if bytes.Equal(original, edited) {
-			if len(results.edit) > 0 {
-				preservedFile(nil, file, out)
-			} else {
-				os.Remove(file)
+			glog.V(4).Infof("User edited:\n%s", string(edited))
+			fmt.Printf("User edited:\n%s", string(edited))
+			lines, err := hasLines(bytes.NewBuffer(edited))
+			if err != nil {
+				return preservedFile(err, file, out)
 			}
-			fmt.Fprintln(out, "Edit cancelled, no changes made.")
-			return nil
-		}
-		if !lines {
-			if len(results.edit) > 0 {
-				preservedFile(nil, file, out)
-			} else {
-				os.Remove(file)
+			if bytes.Equal(original, edited) {
+				if len(results.edit) > 0 {
+					preservedFile(nil, file, out)
+				} else {
+					os.Remove(file)
+				}
+				fmt.Fprintln(out, "Edit cancelled, no changes made.")
+				return nil
+			}
+			if !lines {
+				if len(results.edit) > 0 {
+					preservedFile(nil, file, out)
+				} else {
+					os.Remove(file)
+				}
+				fmt.Fprintln(out, "Edit cancelled, saved file was empty.")
+				return nil
 			}
-			fmt.Fprintln(out, "Edit cancelled, saved file was empty.")
-			return nil
-		}
 
-		results = editResults{
-			file: file,
-		}
+			results = editResults{
+				file: file,
+			}
 
-		// parse the edited file
-		updates, err := rmap.InfoForData(edited, "edited-file")
-		if err != nil {
-			return preservedFile(err, file, out)
-		}
+			// parse the edited file
+			updates, err := rmap.InfoForData(edited, "edited-file")
```

Comment:
When we edit a list of objects (`edited` is a `[]byte` type List object), `InfoForData` ignores the changes users made in each Item of the List, and therefore the changes are never patched when you edit a list. So instead, I parse `infos` as []runtime.Object in L148 and the users now need to traverse/edit them one by one (not editing the List directly). 
@smarterclayton Do you also see this in Openshift origin's edit?